### DISCUSSION TITLE
Use "bundler" as default setting for TypeScript moduleResolution

### DISCUSSION THREAD

**Discussion by ngbrown** [remix-run/remix#6950] [Proposals]
The current default of "node" is really for node 11 and earlier. The "node" setting doesn't work correctly with packages using [subpath exports](https://nodejs.org/api/packages.html#subpath-exports).

Having the default `tsconfig.json` setting `"moduleResolution": "node16"` or `"moduleResolution": "bundler"` (new in TypeScript v5) will get new users on the right track sooner.

See https://github.com/microsoft/TypeScript/issues/50794 and https://devblogs.microsoft.com/typescript/announcing-typescript-5-0/#moduleresolution-bundler for more details.

**Comment by sergiodxa** [comment]
I recently have to change `moduleResolution` from `node` to `bundler` to allow imports of remix-hono to work (it uses package.json exports key), and at the same time not be forced to include the extension of the files on every import.

Having this changed in all templates would be great!

**Comment by ngbrown** [comment]
Fixed in #6949
